tests/unittests/test_exception_handling.py

```diff
@@ -0,0 +1,175 @@
+import unittest
+from unittest import mock
+import json
+import requests
+from parameterized import parameterized
+from tap_freshdesk import client
+from tap_freshdesk.client import raise_for_error, ERROR_CODE_EXCEPTION_MAPPING
+from tap_freshdesk.streams import Tickets, TimeEntries
+
+def get_response(status_code, json_resp={}, headers = None):
+    """
+    Returns mock response
+    """
+    response = requests.Response()
+    response.status_code = status_code
+    response._content = json.dumps(json_resp).encode()
+    if headers:
+        response.headers = headers
+    return response
+
+
+class TestExceptionHanfling(unittest.TestCase):
+    """
+    Test Error is thrown with the expected error message.
+    """
+
+    @parameterized.expand([
+        (400, client.FreshdeskValidationError),
+        (401, client.FreshdeskAuthenticationError),
+        (403, client.FreshdeskAccessDeniedError),
+        (404, client.FreshdeskNotFoundError),
+        (405, client.FreshdeskMethodNotAllowedError),
+        (406, client.FreshdeskUnsupportedAcceptHeaderError),
+        (409, client.FreshdeskConflictingStateError),
+        (415, client.FreshdeskUnsupportedContentError),
+        (429, client.FreshdeskRateLimitError),
+        (500, client.FreshdeskServerError),
+        (503, client.Server5xxError),  # Unknown 5xx error
+    ])
+    def test_custom_error_message(self, error_code, error):
+        """
+        Test that error is thrown with the custom error message
+        if no description is provided in response.
+        """
+        expected_message = "HTTP-error-code: {}, Error: {}".format(error_code, ERROR_CODE_EXCEPTION_MAPPING.get(error_code, {}).get("message", "Unknown Error"))
+        with self.assertRaises(error) as e:
+            raise_for_error(get_response(error_code))
+
+        # Verify that an error message is expected
+        self.assertEqual(str(e.exception), expected_message)
+
+    @parameterized.expand([
+        (400, "Client or Validation Error", client.FreshdeskValidationError),
+        (401, "Authentication Failure", client.FreshdeskAuthenticationError),
+        (403, "Access Denied", client.FreshdeskAccessDeniedError),
+        (404, "Requested Resource not Found", client.FreshdeskNotFoundError),
+        (405, "Method not allowed", client.FreshdeskMethodNotAllowedError),
+        (406, "Unsupported Accept Header", client.FreshdeskUnsupportedAcceptHeaderError),
+        (409, "AInconsistent/Conflicting State", client.FreshdeskConflictingStateError),
+        (415, "Unsupported Content-type", client.FreshdeskUnsupportedContentError),
+        (429, "Rate Limit Exceeded", client.FreshdeskRateLimitError),
+        (500, "Unexpected Server Error", client.FreshdeskServerError),
```

Comment:
Take one un expected error as well like 503 or 505

Reply:
Added test case to verify `503` status code.